app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/DateOfBirthController.scala

```diff
@@ -66,39 +66,35 @@ class DateOfBirthController @Inject()(
         checkSessionStateAndBusinessType(existingSession, agent) { businessType =>
           if (businessType == Llp)
             existingSession.lastNameFromCid match {
-              case Some(_) => Ok(dateOfBirthTemplate(getForm(existingSession.dateOfBirth)))
+              case Some(_) => Ok(dateOfBirthTemplate(getForm(existingSession.dateOfBirth), businessType))
               case None    => Redirect(routes.BusinessIdentificationController.showNoMatchFound())
-            } else Ok(dateOfBirthTemplate(getForm(existingSession.dateOfBirth)))
+            } else Ok(dateOfBirthTemplate(getForm(existingSession.dateOfBirth), businessType))
         }
       }
     }
   }
 
   def submitDateOfBirthForm(): Action[AnyContent] = Action.async { implicit request =>
-    withSubscribingAgent { _ =>
-      dateOfBirthForm
-        .bindFromRequest()
-        .fold(
-          formWithErrors => Ok(dateOfBirthTemplate(formWithRefinedErrors(formWithErrors))),
-          validDob => {
-            sessionStoreService.fetchAgentSession.flatMap {
-              case Some(existingSession) =>
-                existingSession.nino match {
```

Comment:
is there any reason why you removed the nino check in your changes ?

Reply:
the presence of a nino is checked in `checkSessionStateAndBusinessType` -- if not present (for all businessTypes requiring nino) redirect to show nino page